apps/dolly-frontend/src/main/js/src/components/fagsystem/pdlf/form/partials/adresser/bostedsadresse/Bostedsadresse.tsx

```diff
@@ -21,19 +21,55 @@ import { VegadresseVelger } from '~/components/fagsystem/pdlf/form/partials/adre
 import { MatrikkeladresseVelger } from '~/components/fagsystem/pdlf/form/partials/adresser/adressetyper/MatrikkeladresseVelger'
 import { FormikProps } from 'formik'
 import { BestillingsveilederContext } from '~/components/bestillingsveileder/Bestillingsveileder'
+import { DatepickerWrapper } from '~/components/ui/form/inputs/datepicker/DatepickerStyled'
+import { Adressetype } from '~/components/fagsystem/pdlf/PdlTypes'
 
 interface BostedsadresseValues {
 	formikBag: FormikProps<{}>
 }
 
+type BostedsadresseFormValues = {
+	formikBag: FormikProps<{}>
+	path: string
+	idx?: number
+	identtype?: string
+}
+
 type Target = {
 	value: string
 }
 
-export const Bostedsadresse = ({ formikBag }: BostedsadresseValues) => {
+export const BostedsadresseForm = ({
+	formikBag,
+	path,
+	idx,
+	identtype,
+}: BostedsadresseFormValues) => {
+	useEffect(() => {
+		formikBag.setFieldValue(`${path}.adresseIdentifikatorFraMatrikkelen`, undefined)
+		const boadresse = _get(formikBag.values, path)
+		if (_get(boadresse, 'vegadresse') && _get(boadresse, 'vegadresse') !== null) {
+			formikBag.setFieldValue(`${path}.adressetype`, Adressetype.Veg)
+		} else if (
+			_get(boadresse, 'matrikkeladresse') &&
+			_get(boadresse, 'matrikkeladresse') !== null
+		) {
+			formikBag.setFieldValue(`${path}.adressetype`, Adressetype.Matrikkel)
+		} else if (
+			_get(boadresse, 'utenlandskAdresse') &&
+			_get(boadresse, 'utenlandskAdresse') !== null
+		) {
+			formikBag.setFieldValue(`${path}.adressetype`, Adressetype.Utenlandsk)
+		} else if (_get(boadresse, 'ukjentBosted') && _get(boadresse, 'ukjentBosted') !== null) {
+			formikBag.setFieldValue(`${path}.adressetype`, Adressetype.Ukjent)
+		}
+	}, [])
+
+	const valgtAdressetype = _get(formikBag.values, `${path}.adressetype`)
+
 	const opts = useContext(BestillingsveilederContext)
 	const getAdresseOptions = () => {
-		if (opts.identtype && opts.identtype !== 'FNR') {
+		if ((opts?.identtype && opts?.identtype !== 'FNR') || (identtype && identtype !== 'FNR')) {
```

Comment:
Kan dette forenkles til `if (opts?.identtype !== 'FNR' ||  identtype !== 'FNR')` eller tenker jeg feil?

Reply:
Vi må nok ha den slik. For i de fleste tilfeller har man enten opts.identtype ELLER identtype, og vi vil kun sjekke på den av dem som eksisterer. Hvis ikke vil vi alltid få false på sjekken 😊